envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/synchronization/GlobalServiceChanges.kt

```diff
@@ -13,7 +14,14 @@ class GlobalServiceChanges(
     fun combined(): Flux<List<LocalityAwareServicesState>> {
         val serviceStatesStreams: List<Flux<Set<LocalityAwareServicesState>>> = serviceChanges.map { it.stream() }
 
-        return Flux.combineLatest(serviceStatesStreams) { statesArray ->
+        return Flux.combineLatest(
+            serviceStatesStreams.map {
+                // if a number of items emitted by one source is very high, combineLatest may entirely ignore items
+                // emitted by other sources. publishOn with multithreaded scheduler prevents it.
+                it.publishOn(Schedulers.elastic(), 1)
```

Comment:
Is this scheduler shared? (i.e. if I create another from "elastic" pool does it share resources?) Maybe we need something dedicated and probably we should add name so we can monitor it.

Reply:
Yes, it's shared scheduler. We can create dedicated scheduler with `Schedulers.newElastic("name")`.  I didn't do it because I feel like it's not needed. 
But I have no strong opinion on this topic. 
We can measure this scheduler and in fact I added metrics for it to our dashboard.

If anyone have an opinion on this topic LMK.

Reply:
ok, I changed it to dedicated and named scheduler.